tests/unit/aspect.ts

```diff
@@ -171,6 +171,41 @@ registerSuite({
 			}
 		},
 
+		'.on()': {
+			'advising function returns undefined, returns original result'() {
+				const aspectStub1 = sinon.stub();
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 1);
+				assert.isTrue(aspectStub1.calledAfter(methodSpy));
+			},
+
+			'advising function returns defined values, returns advising function result'() {
+				const aspectStub1 = sinon.stub().returns(2);
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 2);
```

Comment:
> IMO, if someone wants to modify the return value, they should be using aspect.after and aspect.on should be simply about notification of the method running (like an event).

In fact, I don't like that it implies that it resolve synchronously too.  I would almost want to remove this feature to keep people from doing stupid stuff.  I would almost be tempted at this point to make `.on` resolve as a scheduled microtask at the end of the turn and @kfranqueiro has an interesting point... someone coming along doing an `on` before an `after` would have a rather surprising unintuitive outcome, especially if the main way of modifying method behaviour will be aspect advice.

While this PR reflects current functionality, I think we should consider the overall functionality now before we regret it...